server/middlewares/user-right.ts

```diff
@@ -20,8 +20,34 @@ function ensureUserHasRight (userRight: UserRight) {
   }
 }
 
+async function ensureUserCanManageChannel (req: express.Request, res: express.Response, next: express.NextFunction) {
```

Comment:
Put this function in `server/middlewares/validators/users` instead

I think you can remove `ensureAuthUserOwnsChannelValidator` because all calls can be replaced by this new function